### PR TITLE
Add Python 3 support, and fix data reception

### PR DESCRIPTION
**Pull request by adelnoureddine**

### CODE REVIEW

File: powerspy.py [hunk 1]
```diff
@@ -135,8 +141,10 @@ def checkID(self):
     s = self.recvCmd(23)
     mat = re.match('POWERSPY(.)(.{12})', s)
     if not mat:
-      # Unable to process response for ID
-      return False
+      mat = re.match('POWERSPY(.)(.{12})', s)
```
Comment (patrickmarlier) [inline powerspy.py]: Why checking the same thing twice?

File: powerspy.py [hunk 2]
```diff
@@ -410,8 +418,9 @@ def exit_gracefully(signal, frame):
   #parser.add_argument('--version', action='version', version='%(prog)s unreleased')
   args = parser.parse_args()
 
-  if args.verbose > 0:
-    logging.basicConfig(level=logging.DEBUG)
+  if args.verbose:
```
Comment (patrickmarlier) [inline powerspy.py]: Revert this.

File: powerspy.py [hunk 3]
```diff
@@ -127,6 +131,8 @@ def recvCmd(self, size = 1):
         buf = mat.group(1)
         logging.debug("RECV: <%s>" % buf)
         break
+    buf = buf.replace("'b'", "") # Fix receiving 'b' on some powerspy connections
```
Comment (patrickmarlier) [inline powerspy.py]: That's strange. I don't have access to PowerSpy device so I will trust you on this.
Maybe add an example of what you observed in the simulator.

**Review by patrickmarlier — CHANGES_REQUESTED** [review]
Thanks for the PR!
Would you mind to split into 2 commits?
One for codecs change and one for the 'b' removal.
Also 2 minor things to change.

File: powerspy.py (hunk 2 above)
Comment (adelnoureddine) [reply powerspy.py]: In Python 3, we can't compare an int with an undefined type. I got this error when I run the script without adding ```-v``` : ```TypeError: '>' not supported between instances of 'NoneType' and 'int'```
So I added this check first. But there might be a more elegant way to do it maybe?

File: powerspy.py (hunk 3 above)
Comment (adelnoureddine) [reply powerspy.py]: Most of the time, the received buffer from recvCmd gives me 'b' interlaced in between the data. For example: ```DEBUG:root:RECV: <P'b'O'b'W'b'E'b'R'b'S'b'P'b'Y'b'C'b'0'b'1'b'0'b'0'b'4'b'0'b'1'b'0'b'0'b'0'b'0'b'1'b'>```
Hence, why I just added the simple replace.

File: powerspy.py (hunk 1 above)
Comment (adelnoureddine) [reply powerspy.py]: Oops, that's a typo, I'll revert this.

**Comment by adelnoureddine** [comment]
> Thanks for the PR!
> Would you mind to split into 2 commits?
> One for codecs change and one for the 'b' removal.
> Also 2 minor things to change.

Thanks Patrick. I'll submit 2 new commits and make a new PR.

**Comment by adelnoureddine** [comment]
I submitted a new PR based on your comments and suggestions.

File: powerspy.py (hunk 2 above)
Comment (patrickmarlier) [reply powerspy.py]: in `add_argument` for verbose, let's add "default=0".

File: powerspy.py (hunk 3 above)
Comment (patrickmarlier) [reply powerspy.py]: That's weird. Are you sure this is not due to python3 here?

**Comment by patrickmarlier** [comment]
You didn't have to open a new PR, just need to force-push your branch and you should get the PR updated with the new history.

File: powerspy.py (hunk 3 above)
Comment (adelnoureddine) [reply powerspy.py]: I'm having it in Python 2 also on Linux.

File: powerspy.py (hunk 2 above)
Comment (adelnoureddine) [reply powerspy.py]: Done.

**Pull request closed without merging**